rebasehelper/tests/test_conf.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+#
+# This tool helps you to rebase package to the latest version
+# Copyright (C) 2013-2014 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# he Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+import os
+import pytest
+import six
+
+from rebasehelper.cli import CLI
+from rebasehelper.config import Conf
+
+
+class TestConfig(object):
+    path_to_conf = os.path.abspath('test_conf.cfg')
+
+    @pytest.mark.parametrize('args', [
+        {
+            'changelog-entry': 'Updated to',
+            'versioneer': 'pypi',
+         },
+        {},
+    ], ids=[
+        'filled',
+        'empty',
+    ])
+    def test_get_config(self, args):
+        self.create_config(args, self.path_to_conf)
+        conf = Conf(self.path_to_conf)
+        expected_result = {key.replace('-', '_'): value for key, value in six.iteritems(args)}
+        assert expected_result == conf.config
+
+        os.remove(self.path_to_conf)
+
+    @pytest.mark.parametrize('cli_args, conf_args, merged', [
+        (['--changelog-entry', 'Version set to',
+          '--buildtool', 'rpmbuild', ],
+         {
+             'changelog-entry': 'Updated to ',
+             'versioneer': 'pypi',
+         },
+         {
+             'changelog-entry': 'Version set to',
+             'versioneer': 'pypi',
+             'buildtool': 'rpmbuild',
+             'color': 'auto',
+         }), ], ids=[
+            'CLI with config',
+        ]
+    )
+    def test_merge(self, cli_args, conf_args, merged):
+        self.create_config(conf_args, self.path_to_conf)
+        expected_output = {x.replace('-', '_'): merged[x] for x in merged}
+        cli = CLI(cli_args)
+        conf = Conf(self.path_to_conf)
+        conf.merge(cli)
+        for key, value in six.iteritems(expected_output):
+            assert conf.config[key] == value
+
+        os.remove(self.path_to_conf)
+
+    @staticmethod
+    def create_config(args, path_to_conf):
```

Comment:
This would be better to implement as a fixture, see this for inspiration: https://github.com/rebase-helper/rebase-helper/blob/master/rebasehelper/tests/test_archive.py#L51-L56

Also, you don't have to remove the file, as it exists in temporary directory.